jib-core/src/main/java/com/google/cloud/tools/jib/cache/DefaultCacheStorageWriter.java

```diff
@@ -204,6 +243,42 @@ CachedLayer write(UncompressedCacheWrite uncompressedCacheWrite) throws IOExcept
     }
   }
 
+  /**
+   * Saves the manifest and container configuration for a V2.2 or OCI image.
+   *
+   * @param imageReference the image reference to store the metadata for
+   * @param manifestTemplate the manifest
+   * @param containerConfiguration the container configuration
+   */
+  void writeMetadata(
+      ImageReference imageReference,
+      BuildableManifestTemplate manifestTemplate,
+      ContainerConfigurationTemplate containerConfiguration)
+      throws IOException {
+    Preconditions.checkNotNull(manifestTemplate.getContainerConfiguration());
+    Preconditions.checkNotNull(manifestTemplate.getContainerConfiguration().getDigest());
+
+    Path imageDirectory = defaultCacheStorageFiles.getImageDirectory(imageReference);
+    Files.createDirectories(imageDirectory);
+
+    writeMetadata(manifestTemplate, imageDirectory.resolve("manifest.json"));
+    writeMetadata(containerConfiguration, imageDirectory.resolve("config.json"));
```

Comment:
Oh! Sorry, missed this important issue. These two writes are sequential, so we need to guard these two with a single lock file so that they are atomic. Directory creation is fine without syncing though, but not sure.